src/ucp/core/ucp_worker.c

```diff
@@ -620,6 +579,53 @@ void ucp_worker_iface_event(int fd, void *arg)
     ucp_worker_signal_internal(worker);
 }
 
+static void ucp_worker_close_ifaces(ucp_worker_h worker)
+{
+    ucp_rsc_index_t rsc_index;
+    ucp_worker_iface_t *wiface;
+    ucp_worker_iface_t *offload_iface;
+    ucs_status_t status;
+
+    UCS_ASYNC_BLOCK(&worker->async);
+
+    for (rsc_index = 0; rsc_index < worker->context->num_tls; ++rsc_index) {
+        wiface = &worker->ifaces[rsc_index];
+        if (wiface->iface == NULL) {
+            continue;
+        }
+
+        uct_worker_progress_unregister_safe(worker->uct,
+                                            &wiface->check_events_id);
+
+        if (wiface->on_arm_list) {
+            ucp_worker_wakeup_ctl_fd(worker, EPOLL_CTL_DEL, wiface->event_fd);
+            ucs_list_del(&wiface->arm_list);
+        }
+
+        if (wiface->attr.cap.flags & UCP_WORKER_UCT_ALL_EVENT_CAP_FLAGS) {
+            status = ucs_async_remove_handler(wiface->event_fd, 1);
+            if (status != UCS_OK) {
+                ucs_warn("failed to remove event handler for fd %d: %s",
+                         wiface->event_fd, ucs_status_string(status));
+            }
+        }
+
+        if (ucp_worker_is_tl_tag_offload(worker, rsc_index)) {
+            ucs_queue_remove(&worker->context->tm.offload.ifaces, &wiface->queue);
+            if (ucp_context_tag_offload_enable(worker->context)) {
+                offload_iface = ucs_queue_head_elem_non_empty(&worker->context->tm.offload.ifaces,
+                                                              ucp_worker_iface_t, queue);
+                ucp_worker_iface_activate(offload_iface);
```

Comment:
shouldn't this be deactivate?

Reply:
No, the intent was the following:

If we have two offload ifaces on different workers, offload is disabled. Then when one of the workers is being destroyed we check whether we can enable offload (if only one proper iface left).

Reply:
so lines 616-618 could be moved inside ucp_context_tag_offload_enable() - i think it would make more sense - offload enable also would activate the offload interface

Reply:
Ok, I’d need to make ucp_worker_iface_activate non static (to be available for ucp_context_tag_offload_enable which is context function). 
Will update PR